Bonsai.Core/WorkflowBuilder.cs

```diff
@@ -888,6 +901,29 @@ static void WriteXmlAttributes(XmlReader reader, XmlWriter writer, bool lookupTy
                                         value = XmlConvert.EncodeName(typeName);
                                     }
                                 }
+                                else if (includeWorkflow &&
+                                         reader.GetAttribute(nameof(IncludeWorkflowBuilder.Path)) is string path &&
+                                         !reader.BaseURI.StartsWith(IncludeWorkflowBuilder.BuildUriPrefix) &&
+                                         visitedWorkflows.Add(path))
+                                {
+                                    // we don't want to fail in most cases while reading nested metadata, as this
+                                    // is an optional performance optimization and we would lose the visual context
+                                    // as to where exactly in the workflow the failure is happening
```

Comment:
Is this an actual concern or just being overly defensive since this is a pretty low-level change to a fundamental system?

Reply:
It's mostly a user-experience thing. It is definitely possible for workflows to have references to missing files or links to files which are actually corrupt, etc. The main issue is that if we let the loader of the main workflow crash when trying to inspect those files, we will have no way of telling visually in which part of the internal workflow the reference is located (e.g. it could be located in a particular node two-levels deep).

This is because the eager lookahead is done without actually loading up and deserializing the XML, so we don't know about internal structure at this point. If there is an exception like this, it will also crash later when the included workflow is actually built and at that time the same exception will be thrown but at the right place in the workflow.